packages/bot-skeleton/src/scratch/hooks/block_svg.js

```diff
@@ -67,10 +68,12 @@ Blockly.BlockSvg.prototype.showContextMenu_ = function (e) {
     // Save the current block in a variable for use in closures.
     const block = this;
     const menu_options = [];
+    if (!config.mainBlocks.includes(block.type) && this.isDeletable() && this.isMovable() && !block.isInFlyout) {
```

Comment:
we can remove the first check `!config.mainBlocks.includes(block.type)`, since main block don't have parents for which we have a check below at line 193